packages/web-components/src/components/va-progress-bar/va-progress-bar.tsx

```diff
@@ -0,0 +1,65 @@
+import { Component, Event, EventEmitter, Host, Prop, h } from '@stencil/core';
+
+@Component({
+  tag: 'va-progress-bar',
+  styleUrl: 'va-progress-bar.css',
+  shadow: true,
+})
+export class VaProgressBar {
+  /**
+   * Whether or not an analytics event will be fired.
+   */
+  @Prop() enableAnalytics: boolean;
+
+  /**
+   * The text label for the progress bar.
+   */
+  @Prop() label: string;
+
+  /**
+   * Percent of progress made. 0 to 100.
+   */
+  @Prop() percent: number;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  componentDidRender() {
+    if (this.enableAnalytics && (this.percent === 0 || this.percent === 100)) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-progress-bar',
+        action: 'change',
+        details: {
+          label: this.label,
+          percent: this.percent,
+        },
+      });
+    }
+  }
+
+  render() {
+    const ariaLabel = this.label ?? `${this.percent}%`;
```

Comment:
Looks like this is causing a test to fail.

Also, instead of nullish coalescing, when we're going for default values can we set that on the property itself, or use destructuring with default assignment?

Reply:
I've pushed up a fix for the tests. I think the default value that we want for label is the value of the percent prop, but we expose a label prop to allow a developer to override it.

Reply:
How about this?

```
  componentWillLoad() {
    if (!this.label) this.label = `${this.percent}`;
  }
```

Reply:
I'm against the previous componentWillLoad idea after implementing it locally because it's adding unnecessary complexity.

Reply:
What about something like this, but "complete" instead of "Working"? This diff is a bit awkward because I moved `percent` above `label` :upside_down_face: 

![image](https://user-images.githubusercontent.com/2008881/145312608-6855847c-cdce-4790-805f-088a90b74e06.png)

Reply:
It looks like this approach doesn't work.
<img width="718" alt="Screen Shot 2021-12-09 at 09 47 43" src="https://user-images.githubusercontent.com/36863582/145418182-e70d90ec-4c56-4846-b922-ee24ac476351.png">

Reply:
I've updated it to use destructuring with default assignment.